flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/ReducingUpsertWriterTest.java

```diff
@@ -382,4 +543,33 @@ public RowData next() {
             }
         }
     }
+
+    private class ReusableIteratorWithNullTimestamp implements Iterator<RowData> {
```

Comment:
I think the iterator is overkill here. You can even test this functionality with one record. Basically you want to confirm if a record with a null timestamp can be flushed.